golang/cosmos/x/swingset/types/msgs_test.go

```diff
@@ -88,3 +90,135 @@ func TestWalletSpendAction(t *testing.T) {
 		})
 	}
 }
+
+func TestInstallBundle_ValidateBasic(t *testing.T) {
+	for _, tt := range []struct {
+		name      string
+		msg       *MsgInstallBundle
+		shouldErr bool
+	}{
+		{
+			name:      "empty",
+			msg:       &MsgInstallBundle{},
+			shouldErr: true,
+		},
+		{
+			name:      "no bundle",
+			msg:       NewMsgInstallBundle("", addr),
+			shouldErr: true,
+		},
+		{
+			name: "uncompressed",
+			msg:  NewMsgInstallBundle("true", addr),
+		},
+		{
+			name: "compressed",
+			msg: &MsgInstallBundle{
+				Submitter:        addr,
+				CompressedBundle: []byte{1, 2, 3},
+				UncompressedSize: 4,
+			},
+		},
+		{
+			name: "both",
+			msg: &MsgInstallBundle{
+				Bundle:           "foo",
+				Submitter:        addr,
+				CompressedBundle: []byte{1},
+			},
+			shouldErr: true,
+		},
+		{
+			name: "zero size",
+			msg: &MsgInstallBundle{
+				Submitter:        addr,
+				CompressedBundle: []byte{1, 2, 3},
+			},
+			shouldErr: true,
+		},
+		{
+			name: "negative size",
+			msg: &MsgInstallBundle{
+				Submitter:        addr,
+				CompressedBundle: []byte{1, 2, 3},
+				UncompressedSize: -4,
+			},
+			shouldErr: true,
+		},
```

Comment:
```suggestion
		{
			name: "negative size",
			msg: &MsgInstallBundle{
				Submitter:        addr,
				CompressedBundle: []byte{1, 2, 3},
				UncompressedSize: -4,
			},
			shouldErr: true,
		},
		{
			name: "oversized",
			msg: &MsgInstallBundle{
				Submitter:        addr,
				CompressedBundle: []byte{1, 2, 3},
				UncompressedSize: math.MaxUint64,
			},
			shouldErr: true,
		},
```

Reply:
This is actually fine as far as `ValidateBasic()` is concerned - for all it knows those 3 bytes uncompress to that huge size. However, I'll add some cases to the other test to make sure it fails if the uncompressed size is modified either up or down.

Reply:
Ah, I see the point of checking for Max. (It should be for MaxInt64 to match the types for `io.LimitedReader`, `io.Copy()`, etc.) Done.